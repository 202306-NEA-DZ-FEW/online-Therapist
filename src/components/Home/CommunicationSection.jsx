import React from "react";

import CommunicationCard from "../Cards/CommunicationCard";
import { useTranslation } from "next-i18next";

function CommunicationSection() {
    const { t } = useTranslation("common");
    return (
        <div className=' p-16 bg-white'>
            <h2 className='lg:text-5xl md:text-3xl text-3xl text-black font-bold uppercase text-center font-atkinson'>
                {t("communication.heading")}
            </h2>
            <div className='flex justify-center mx-auto mt-12 mr-16 h-auto '>
                <div className='flex-1 overflow-y-auto sm:pl-16'>
                    <div className='flex flex-wrap justify-center bg m-4 md:flex-row md:gap-8 gap-4'>
                        <CommunicationCard
                            greenBackground='bg-LightTeal'
                            icon='call' // Use 'call' for the phone icon
                            title={t("communication.voiceCall")}
                            paragraph= {t("communication.voiceText")}
                        />

                        <CommunicationCard
                            greenBackground='bg-SkyBlue'
                            icon='videoCall' // Use 'videoCall' for the video call icon
                            title={t("communication.videoCall")}
                            paragraph= {t("communication.videoText")}
                        />

                        <CommunicationCard
                            greenBackground='bg-LightOrange'
                            icon='chat' // Use 'chat' for the chat icon
                            title= {t("communication.chat")}
                            paragraph= {t("communication.chatText")}
                        />
                    </div>
                </div>
            </div>
        </div>
    );
}

export default CommunicationSection;
