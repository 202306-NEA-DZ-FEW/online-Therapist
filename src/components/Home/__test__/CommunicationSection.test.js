import renderer from "react-test-renderer";

import CommunicationSection from "../CommunicationSection";

it("renders correctly", () => {
    const tree = renderer.create(<CommunicationSection />).toJSON();
    expect(tree).toMatchSnapshot();
});
