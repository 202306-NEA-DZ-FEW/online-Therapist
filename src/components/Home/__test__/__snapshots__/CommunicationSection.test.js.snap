// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className=" p-16 bg-white"
>
  <div
    style={
      Object {
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <div
      style={
        Object {
          "opacity": 0,
          "transform": "translateY(75px) translateZ(0)",
        }
      }
    >
      <h2
        className="lg:text-5xl md:text-3xl text-3xl text-black font-bold uppercase text-center font-atkinson"
      >
        communication.heading
      </h2>
    </div>
  </div>
  <div
    className="flex justify-center mx-auto mt-12 mr-16 h-auto "
  >
    <div
      className="flex-1 overflow-y-auto sm:pl-16"
    >
      <div
        className="flex flex-wrap justify-center bg m-4 md:flex-row md:gap-8 gap-4"
      >
        <div
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "opacity": 0,
                "transform": "translateY(75px) translateZ(0)",
              }
            }
          >
            <div
              className="h-auto lg:h-[355px]"
            >
              <div
                className="hover:scale-105 border border-gray p-1 h:44 w-56 lg:h-80 lg:w-60 rounded-3xl m-4 shadow-xl"
              >
                <div
                  className="flex flex-col h-44 w-[2356x] rounded-[20px]  bg-LightTeal "
                >
                  <span
                    className="text-6xl m-auto"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="2"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M15.05 5A5 5 0 0 1 19 8.95M15.05 1A9 9 0 0 1 23 8.94m-1 7.98v3a2 2 0 0 1-2.18 2 19.79 19.79 0 0 1-8.63-3.07 19.5 19.5 0 0 1-6-6 19.79 19.79 0 0 1-3.07-8.67A2 2 0 0 1 4.11 2h3a2 2 0 0 1 2 1.72 12.84 12.84 0 0 0 .7 2.81 2 2 0 0 1-.45 2.11L8.09 9.91a16 16 0 0 0 6 6l1.27-1.27a2 2 0 0 1 2.11-.45 12.84 12.84 0 0 0 2.81.7A2 2 0 0 1 22 16.92z"
                      />
                    </svg>
                  </span>
                  <h3
                    className="mx-auto text-3xl font-atkinson font- mb-8"
                  >
                    communication.voiceCall
                  </h3>
                </div>
                <p
                  className="font-atkinson text-center p-6 text-lg text-gray-700"
                >
                  communication.voiceText
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "opacity": 0,
                "transform": "translateY(75px) translateZ(0)",
              }
            }
          >
            <div
              className="h-auto lg:h-[355px]"
            >
              <div
                className="hover:scale-105 border border-gray p-1 h:44 w-56 lg:h-80 lg:w-60 rounded-3xl m-4 shadow-xl"
              >
                <div
                  className="flex flex-col h-44 w-[2356x] rounded-[20px]  bg-SkyBlue "
                >
                  <span
                    className="text-6xl m-auto"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="2"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <polygon
                        points="23 7 16 12 23 17 23 7"
                      />
                      <rect
                        height="14"
                        rx="2"
                        ry="2"
                        width="15"
                        x="1"
                        y="5"
                      />
                    </svg>
                  </span>
                  <h3
                    className="mx-auto text-3xl font-atkinson font- mb-8"
                  >
                    communication.videoCall
                  </h3>
                </div>
                <p
                  className="font-atkinson text-center p-6 text-lg text-gray-700"
                >
                  communication.videoText
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          style={
            Object {
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <div
            style={
              Object {
                "opacity": 0,
                "transform": "translateY(75px) translateZ(0)",
              }
            }
          >
            <div
              className="h-auto lg:h-[355px]"
            >
              <div
                className="hover:scale-105 border border-gray p-1 h:44 w-56 lg:h-80 lg:w-60 rounded-3xl m-4 shadow-xl"
              >
                <div
                  className="flex flex-col h-44 w-[2356x] rounded-[20px]  bg-LightOrange "
                >
                  <span
                    className="text-6xl m-auto"
                  >
                    <svg
                      fill="none"
                      height="1em"
                      stroke="currentColor"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth="2"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M14 9a2 2 0 0 1-2 2H6l-4 4V4c0-1.1.9-2 2-2h8a2 2 0 0 1 2 2v5Z"
                      />
                      <path
                        d="M18 9h2a2 2 0 0 1 2 2v11l-4-4h-6a2 2 0 0 1-2-2v-1"
                      />
                    </svg>
                  </span>
                  <h3
                    className="mx-auto text-3xl font-atkinson font- mb-8"
                  >
                    communication.chat
                  </h3>
                </div>
                <p
                  className="font-atkinson text-center p-6 text-lg text-gray-700"
                >
                  communication.chatText
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
